backend/app/api/routes/dde.py

```diff
@@ -279,13 +279,61 @@ async def extract_from_file(*, session: SessionDep, current_user: CurrentUser, n
     
     chat_completion_response = await ArenaHandler(session, document_data_extractor.owner, chat_completion_request).process_request()
     extracted_info=chat_completion_response.choices[0].message.content
+    token_info = chat_completion_response.choices[0].logprobs.content
     #TODO: handle refusal or case in which content was not correctly done
     # TODO: Improve the prompt to ensure the output is always a valid JSON
     json_string = extracted_info[extracted_info.find('{'):extracted_info.rfind('}')+1]
-    extracted_data = {k: v for k, v in json.loads(json_string).items() if k not in ('source', 'year')}   
-    logprob_data = extract_logprobs_from_response(chat_completion_response, extracted_data)
+    keys = list(json.loads(json_string).keys())
+    value_indices = find_indices_for_keys(keys, token_info)
```

Comment:
try to use expressive names for function so that one can directly understand what they are doing, like `extract_tokens_indices_for_each_key`